streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -620,6 +596,55 @@ object ZTransducer extends ZTransducerPlatformSpecificConstructors {
   def fromPush[R, E, I, O](push: Option[Chunk[I]] => ZIO[R, E, Chunk[O]]): ZTransducer[R, E, I, O] =
     ZTransducer(Managed.succeed(push))
 
+  /**
+   * Creates a summary from a transducer and a predicate function.
+   */
+  def fromSummary[I, O](summary: Summary[I, O])(emitWhen: O => Boolean): ZTransducer[Any, Nothing, I, O] =
```

Comment:
Consider also adding `toTransducer` on `Summary`.

Reply:
It's there and uses the `Summary[I, Boolean]` scheme we came up with, but I'm still experimenting with different ways of signaling when to emit.

We may still want a way to abstract over these emission conditions.